app/controllers/api/json_controller.rb

```diff
@@ -39,32 +32,46 @@ def json
         @ready_media << @ready_videos
         @ready_media = @ready_media.compact_blank
 
-        @finished_json = build_json
-        response.set_header('Access-Control-Allow-Origin', '*')
-        render json: @finished_json
+      @narratives =  search_narratives(params["search"])
+      @ready_narratives =[]
+
+      @narratives.each{|narrative|  @ready_narratives.append(make_narrative(narrative,params["year"])) } 
+      @ready_narratives = @ready_narratives.compact
+
 
+      @audios =  search_audios(params["search"])
+
+      @ready_media = []
+      @ready_audios =[]
+
+      @audios.each{|audio|  @ready_audios.append(make_audio(audio,params["year"])) } 
+      @ready_audios = @ready_audios.compact_blank
+
+      @ready_media << @ready_audios
```

Comment:
This seems to be overwriting rather than appending to media. As a result, we get the correct count at the end, but the media array does not contain the proper number of elements.